### PR TITLE
Design login

### PR DESCRIPTION
**Pull request by vuducdat**

### CODE REVIEW

File: app/assets/javascripts/sessions.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (leminhtuan2015) [inline app/assets/javascripts/sessions.coffee]: Do you need this file?

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  def new
+  end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      redirect_to user
+    else
+      flash.now[:danger] = "Invalid email/password combination"
```
Comment (leminhtuan2015) [inline app/controllers/sessions_controller.rb]: I18n

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      render "new"
```
Comment (leminhtuan2015) [inline app/controllers/sessions_controller.rb]: :new

File: app/helpers/sessions_helper.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module SessionsHelper
+  def log_in(user)
```
Comment (leminhtuan2015) [inline app/helpers/sessions_helper.rb]: Remove ( )

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -5,7 +5,24 @@
       <ul class="nav navbar-nav navbar-right">
         <li><%= link_to t("title.home2"), root_path %></li>
         <li><%= link_to t("title.help"), help_path %></li>
-        <li><%= link_to t("title.login"), '#' %></li>
+        <% if logged_in? %>
+          <li><%= link_to "Users", '#' %></li>
```
Comment (leminhtuan2015) [inline app/views/layouts/_header.html.erb]: I18n

File: app/assets/stylesheets/sessions.scss [hunk 6]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
```
Comment (leminhtuan2015) [inline app/assets/stylesheets/sessions.scss]: Remove this file.

File: app/views/layouts/_header.html.erb [hunk 7]
```diff
@@ -5,7 +5,24 @@
       <ul class="nav navbar-nav navbar-right">
         <li><%= link_to t("title.home2"), root_path %></li>
         <li><%= link_to t("title.help"), help_path %></li>
-        <li><%= link_to t("title.login"), '#' %></li>
+        <% if logged_in? %>
+          <li><%= link_to t("title.user"), '#' %></li>
+          <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+              Account <b class="caret"></b>
```
Comment (leminhtuan2015) [inline app/views/layouts/_header.html.erb]: I18n

File: app/views/layouts/_header.html.erb [hunk 8]
```diff
@@ -5,7 +5,24 @@
...
+            </a>
+            <ul class="dropdown-menu">
+              <li><%= link_to t("menu.profile"), current_user %></li>
+              <li><%= link_to t("menu.setting"), '#' %></li>
```
Comment (leminhtuan2015) [inline app/views/layouts/_header.html.erb]: ' -> "

File: app/views/layouts/_header.html.erb [hunk 9]
```diff
@@ -5,7 +5,24 @@
...
+              <li class="divider"></li>
+              <li>
+                <%= link_to t("title.logout"), logout_path, method: "delete" %>
+              </li>
+            </ul>
+          </li>
+        <% else %>
+          <li><%= link_to t("title.login"), '#' %></li>
```
Comment (leminhtuan2015) [inline app/views/layouts/_header.html.erb]: ' -> "

File: app/views/layouts/_header.html.erb [hunk 10]
```diff
@@ -5,7 +5,24 @@
       <ul class="nav navbar-nav navbar-right">
         <li><%= link_to t("title.home2"), root_path %></li>
         <li><%= link_to t("title.help"), help_path %></li>
-        <li><%= link_to t("title.login"), '#' %></li>
+        <% if logged_in? %>
+          <li><%= link_to t("title.user"), '#' %></li>
```
Comment (VoTaiTri) [inline app/views/layouts/_header.html.erb]: use " instead '
and same below

File: config/locales/en.yml [hunk 11]
```diff
@@ -33,8 +13,15 @@ en:
...
     contact: "Contact"
     error: "The form contains"
     new: "News"
-
+    login1: "Sign up now!"
+    user: "Users"
+  menu:
+    profile: "Profile"
+    setting: "Settings"
+  danger:
```
Comment (VoTaiTri) [inline config/locales/en.yml]: rename more clean

**Comment by leminhtuan2015** [comment]
LGTM
Good job.

File: app/controllers/sessions_controller.rb [hunk 12]
```diff
@@ -0,0 +1,21 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      params[:session][:remember_me] == '1' ? remember(user) : forget(user)
```
Comment (leminhtuan2015) [inline app/controllers/sessions_controller.rb]: Turn 1 to constant.
Every thing have its meaning

File: app/helpers/sessions_helper.rb [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def forget user
+    user.forget
+    cookies.delete :user_id
+    cookies.delete :remember_token
+  end
+
+  def log_out
+    forget(current_user)
```
Comment (leminhtuan2015) [inline app/helpers/sessions_helper.rb]: remove ( )

File: app/models/user.rb [hunk 14]
```diff
@@ -1,15 +1,17 @@
 class User < ActiveRecord::Base
   has_many :lessons, dependent: :destroy
   has_many :activities, dependent: :destroy
-  has_many :active_relationships,  class_name:  "Relationship",
+  has_many :active_relationships, class_name:  "Relationship",
```
Comment (leminhtuan2015) [inline app/models/user.rb]: Space

File: app/models/user.rb [hunk 15]
```diff
@@ -18,4 +20,23 @@ class User < ActiveRecord::Base
...
+    BCrypt::Password.create(string, cost: cost)
+  end
+
+  def User.new_token
+    SecureRandom.urlsafe_base64
+  end
+
+  def authenticated? remember_token
+    return false if remember_digest.nil? 
+    BCrypt::Password.new(remember_digest).is_password?(remember_token)
```
Comment (leminhtuan2015) [inline app/models/user.rb]: remove ( )

**Comment by leminhtuan2015** [comment]
OK

File: app/models/user.rb [hunk 16]
```diff
@@ -18,4 +20,23 @@ class User < ActiveRecord::Base
   format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false} 
   has_secure_password
   validates :password, presence: true, length: {minimum: 6}
```
Comment (phonghenry14) [inline app/models/user.rb]: cho 6 vào Settings này

**Pull request merged**